lib/processing.py

```diff
@@ -0,0 +1,490 @@
+            if name_similarity > 0.9 and lastname_similarity > 0.9:
+                return {"Name_extracted": highest_name, 
+                        "last_name_extracted": highest_lastname, 
+                        "Bounding_boxes_name": get_matching_bounding_boxes(highest_name, bounding_boxes), 
+                        "Bounding_boxes_last_name": get_matching_bounding_boxes(highest_lastname, bounding_boxes),
+                        "fuzzy_matching_name": name_similarity * 100,
+                        "fuzzy_matching_lastname": lastname_similarity * 100}
+            else:
+                return {"outcome": f"the highest name similarity is: {highest_name_simitality} with the extract name {highest_name}  and the highest lastname similarity is: {highest_lastname_simitality} with the extract lastname {highest_lastname}"}
+        else:
+            answer = get_answer_llm(config, extracted_text)
+            dictionary = convert_str_to_dict(answer)
+            name_similarity = calculate_similarity(dictionary["name"], name)
+            lastname_similarity = calculate_similarity(dictionary["last_name"], lastname)
+            if name_similarity > 0.9 and lastname_similarity > 0.9:
+                return {"Name_extracted": dictionary["name"], 
+                        "last_name_extracted": dictionary["last_name"], 
+                        "Bounding_boxes_name": get_matching_bounding_boxes(dictionary["name"], bounding_boxes), 
+                        "Bounding_boxes_last_name": get_matching_bounding_boxes(dictionary["last_name"], bounding_boxes),
+                        "fuzzy_matching_name": name_similarity * 100,
+                        "fuzzy_matching_lastname": lastname_similarity * 100}
+            else:
+                highest_name_simitality, highest_lastname_simitality, highest_name, highest_lastname, extracted_text, bounding_boxes, name_similarity, lastname_similarity = modify_angle_extract_data(config, file_path, name, lastname, highest_name_simitality, highest_lastname_simitality)
+                if name_similarity > 0.9 and lastname_similarity > 0.9:
+                    return {"Name_extracted": highest_name, 
+                            "last_name_extracted": highest_lastname, 
+                            "Bounding_boxes_name": get_matching_bounding_boxes(highest_name, bounding_boxes), 
+                            "Bounding_boxes_last_name": get_matching_bounding_boxes(highest_lastname, bounding_boxes),
+                            "fuzzy_matching_name": name_similarity * 100,
+                            "fuzzy_matching_lastname": lastname_similarity * 100}
+                else:
+                    return {"outcome": f"the highest name similarity is: {highest_name_simitality} with the extract name {highest_name}  and the highest lastname similarity is: {highest_lastname_simitality} with the extract lastname {highest_lastname}"}
+
+
+## FUNTIONS TO CREATE A RAG 
+def normalize_text(s):
```

Comment:
When using openai embeddings, you can leave some of these special characters, they trained the model to use those and their pre-processor removes whatever they need to be removed, but I like the approach. Same for huggingface, they have their own tokenizers